nilearn/datasets/neurovault.py

```diff
@@ -0,0 +1,2589 @@
+
+    Raises
+    ------
+    URLError, ValueError
+        If an error occurred when downloading the file.
+
+    See Also
+    --------
+    nilearn.datasets._utils._fetch_file
+
+
+    Notes
+    -----
+    It can happen that an HTTP error that occurs inside
+    ``_fetch_file`` gets transformed into an ``AttributeError`` when
+    we try to set the ``reason`` attribute of the exception raised;
+    here we replace it with an ``URLError``.
+
+    """
+    print('Downloading file {} to {}'.format(url, target_file))
+    _logger.debug('Downloading file: {0}'.format(url))
+    try:
+        downloaded = _fetch_file(url, temp_dir, resume=False,
+                                 overwrite=True, verbose=0)
+    except Exception as e:
+        _logger.error('Problem downloading file from {0}'.format(url))
+
+        # reason is a property of urlib.error.HTTPError objects,
+        # but these objects don't have a setter for it, so
+        # an HTTPError raised in _fetch_file might be transformed
+        # into an AttributeError when we try to set its reason attribute
+        if (isinstance(e, AttributeError) and
+            e.args[0] == "can't set attribute"):
+            raise URLError(
+                'HTTPError raised in nilearn.datasets._fetch_file: {0}'.format(
+                    traceback.format_exc()))
+        raise
+    shutil.move(downloaded, target_file)
+    _logger.debug(
+        'Download succeeded, downloaded to: {0}'.format(target_file))
+    return target_file
+
+
+def _checked_get_dataset_dir(dataset_name, suggested_dir=None,
+                             write_required=False):
+    """Wrapper for ``_get_dataset_dir``.
+
+    Expands . and ~ and checks write access.
+
+    Parameters
+    ----------
+    dataset_name : str
+        Passed to ``_get_dataset_dir``. Example: ``neurovault``.
+
+    suggested_dir : str
+        Desired location of root data directory for all datasets,
+        e.g. ``~/home/nilearn_data``.
+
+    write_required : bool, optional (default=False)
+        If ``True``, check that the user has write access to the
+        chosen data directory and raise ``IOError`` if not.  If
+        ``False``, don't check for write permission.
+
+    Returns
+    -------
+    dataset_dir : str
+        The location of the dataset directory in the filesystem.
+
+    Raises
+    ------
+    IOError
+        If `write_required` is set and the user doesn't have write
+        access to `dataset_dir`.
+
+    See Also
+    --------
+    nilearns.datasets._utils._get_dataset_dir
+
+    """
+    if suggested_dir is not None:
+        suggested_dir = os.path.abspath(os.path.expanduser(suggested_dir))
+    dataset_dir = _get_dataset_dir(dataset_name, data_dir=suggested_dir)
+    if not write_required:
+        return dataset_dir
+    if not os.access(dataset_dir, os.W_OK):
+        raise IOError('Permission denied: {0}'.format(dataset_dir))
+    return dataset_dir
+
+
+def neurovault_directory(suggested_dir=None):
```

Comment:
Let's remove this for the user, and let's try to use the same code as in the other downloaders.